### PR TITLE
Enable probot autolabeler

### PR DESCRIPTION
**Pull request by alexvanboxel**
Waiting for INFRA enablement

R: iemejia

### CODE REVIEW

File: .github/autolabeler.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+infra: [".test-infra/*"]
+build: ["buildSrc/*", "release/*", "ownership/*", ".github/*", "gradle/*"]
+website: ["website/*"]
+go: ["sdks/go/*"]
+java: ["sdks/java/*"]
+python: ["sdks/python/*"]
+runner: ["runners/*"]
+examples: ["examples/*", "learning/*"]
+model: ["model/*"]
```
Comment (sunjincheng121) [inline .github/autolabeler.yml]: Add newline at the end of file.

Comment (alexvanboxel) [reply .github/autolabeler.yml]: Added newline

File: .github/autolabeler.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# limitations under the License.
+#
+
+infra: [".test-infra/*"]
+build: ["buildSrc/*", "release/*", "ownership/*", ".github/*", "gradle/*"]
+website: ["website/*"]
+go: ["sdks/go/*"]
+java: ["sdks/java/*"]
+python: ["sdks/python/*"]
+runner: ["runners/*"]
```
Comment (iemejia) [inline .github/autolabeler.yml]: Can we expand this also into the individual runners, e.g.
```
flink: ["runners/flink/*"]
spark: ["runners/spark/*"]
...
```

File: .github/autolabeler.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+examples: ["examples/*", "learning/*"]
```
Comment (iemejia) [inline .github/autolabeler.yml]: I would keep learning separate, these are different things and commonly mainted by different people who may be interesting in filtering one or the other.

**Comment by iemejia** [comment]
I had almost the same suggestion than @sunjincheng121 however there is a small refinement to do over that one, do not put prefixes e.g. `extensions-java-sorter` use better `sorter` the other paths are matched by the other tag rules the specific rule for java and for extensions.

**Comment by iemejia** [comment]
Forgot to mention, Can we sort the elements of the file so it can be easier to maintain too.

**Comment by alexvanboxel** [comment]
I've added @iemejia and @sunjincheng121 suggestions. I did use a prefix though to make it usable in maintaining labels. I hope the auto-labeler can handle them. Prefix used are:

- extensions: ext/ (example: ext/jackson)
- io: io/ (example: io/mongodb)
- runners: run/ (example: run/flink)

I also sorted the labels. Squashed commit.

**Comment by alexvanboxel** [comment]
Added the main labels already to the repo (not the sub labels)

**Comment by iemejia** [comment]
Merged manually to do the extra adjustments. I did not want to bother anymore with those.
Since INFRA apparently enabled the action now is time to see those auto labels appearing :)

**Pull request closed without merging**